canary_tests.py

```diff
@@ -129,6 +128,21 @@ async def record_tick_data(tick_stream, data, should_stop):
             if should_stop.is_set():
                 return
 
+class PingMonitorTest(unittest.IsolatedAsyncioTestCase):
+    async def test_monitor_is_online(self):
+        """
+        One board canary is designated the "canary monitor," and makes sure
+        that all other canaries are online. There is a different canary that's
+        in charge of making sure the canary monitor is online, and that's what
+        this test is about.
+        """
+        if conf.board_monitor is None:
+            return  # No monitor to connect to
+
+        address, creds = conf.board_monitor
+        robot = await RobotClient.at_address(address, creds)
```

Comment:
should we log/print the error here if it fails to connect? or does that happen automatically?

Reply:
My hope is that if it fails to connect, it raises some kind of ConnectionError, and that should get printed/logged when the test fails with uncaught exceptions. I'll double-check...

Reply:
Yup! I tried it with an invalid address:
```
2024-06-27 20:24:50,364		ERROR	viam.robot.client (client.py:257)	Unable to establish a connection to the robot. Ensure the robot is online and reachable and try again.	
E
======================================================================
ERROR: test_monitor_is_online (__main__.PingMonitorTest)
One board canary is designated the "canary monitor," and makes sure
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/viam/robot/client.py", line 255, in _with_channel
    await self.refresh()
  File "/usr/local/lib/python3.10/dist-packages/viam/robot/client.py", line 297, in refresh
    response: ResourceNamesResponse = await self._client.ResourceNames(ResourceNamesRequest())
  File "/usr/local/lib/python3.10/dist-packages/grpclib/client.py", line 880, in __call__
    async with self.open(timeout=timeout, metadata=metadata) as stream:
  File "/usr/local/lib/python3.10/dist-packages/grpclib/client.py", line 563, in __aexit__
    raise exc_val
  File "/usr/local/lib/python3.10/dist-packages/grpclib/client.py", line 553, in __aexit__
    await self._maybe_finish()
  File "/usr/local/lib/python3.10/dist-packages/grpclib/client.py", line 523, in _maybe_finish
    await self.recv_trailing_metadata()
  File "/usr/local/lib/python3.10/dist-packages/grpclib/client.py", line 484, in recv_trailing_metadata
    self._raise_for_grpc_status(status, message, details)
  File "/usr/local/lib/python3.10/dist-packages/grpclib/client.py", line 345, in _raise_for_grpc_status
    raise GRPCError(status, message, details)
grpclib.exceptions.GRPCError: (<Status.UNIMPLEMENTED: 12>, 'unknown service viam.robot.v1.RobotService', None)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.10/unittest/async_case.py", line 64, in _callTestMethod
    self._callMaybeAsync(method)
  File "/usr/lib/python3.10/unittest/async_case.py", line 87, in _callMaybeAsync
    return self._asyncioTestLoop.run_until_complete(fut)
  File "/usr/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/usr/lib/python3.10/unittest/async_case.py", line 101, in _asyncioLoopRunner
    ret = await awaitable
  File "/home/viam/board_canaries/./canary_tests.py", line 143, in test_monitor_is_online
    robot = await RobotClient.at_address(address, creds)
  File "/usr/local/lib/python3.10/dist-packages/viam/robot/client.py", line 197, in at_address
    robot = await cls._with_channel(channel, options, True)
  File "/usr/local/lib/python3.10/dist-packages/viam/robot/client.py", line 259, in _with_channel
    raise ConnectionError("Unable to establish a connection to the robot.")
ConnectionError: Unable to establish a connection to the robot.

----------------------------------------------------------------------
Ran 5 tests in 25.682s

FAILED (errors=1)
```